tools/cldr-apps/src/main/java/org/unicode/cldr/web/auth/GithubLoginFactory.java

```diff
@@ -0,0 +1,201 @@
+
+    private PrivateKey privateKey = null;
+    private PublicKey publicKey = null;
+
+    /** construct a new login factory */
+    public GithubLoginFactory() {
+        CLDRConfig instance = CLDRConfig.getInstance();
+
+        GITHUB_APP_ID = instance.getProperty("GITHUB_APP_ID");
+        GITHUB_CLIENT_ID = instance.getProperty("GITHUB_CLIENT_ID");
+        GITHUB_CLIENT_SECRET = instance.getProperty("GITHUB_CLIENT_SECRET");
+        final File homeFile = ((CLDRConfigImpl) instance).getHomeFile();
+        final Path derPrivateFile = new File(homeFile, PRIVATE_FILE_NAME).toPath();
+        final Path derPublicFile = new File(homeFile, PUBLIC_FILE_NAME).toPath();
+        byte[] derPublic = null;
+        byte[] derPrivate = null;
+        try {
+            derPrivate = Files.readAllBytes(derPrivateFile);
+        } catch (java.io.IOException e) {
+            logger.log(
+                    java.util.logging.Level.SEVERE,
+                    "Could not load " + derPrivateFile + ": " + e.getMessage(),
+                    e);
+        }
+        try {
+            derPublic = Files.readAllBytes(derPublicFile);
+        } catch (java.io.IOException e) {
+            logger.log(
+                    java.util.logging.Level.SEVERE,
+                    "Could not load " + derPublicFile + ": " + e.getMessage(),
+                    e);
+        }
+
+        if (derPublic != null && derPrivate != null) {
+            try {
+                final KeyFactory rsaFactory = KeyFactory.getInstance("RSA");
+                publicKey = rsaFactory.generatePublic(new X509EncodedKeySpec(derPublic));
+                privateKey = rsaFactory.generatePrivate(new PKCS8EncodedKeySpec(derPrivate));
+            } catch (NoSuchAlgorithmException | InvalidKeySpecException ex) {
+                logger.log(
+                        java.util.logging.Level.SEVERE,
+                        "Could not read private/public keys: " + ex.getMessage(),
+                        ex);
+            }
+        }
+    }
+
+    @Override
+    public boolean valid() {
+        if (GITHUB_APP_ID == null
+                || GITHUB_APP_ID.isBlank()
+                || GITHUB_CLIENT_ID == null
+                || GITHUB_CLIENT_ID.isBlank()
+                || GITHUB_CLIENT_SECRET == null
+                || GITHUB_CLIENT_SECRET.isBlank()
+                || privateKey == null
+                || publicKey == null) {
+            return false;
+        }
+
+        return true;
+    }
+
+    // /** Construct the JWT identifying as the client */
+    // private String getClientJwt() {
+    //     final long now = System.currentTimeMillis();
+    //     return Jwts.builder()
+    //             .setSubject("CLDR SurveyTool")
+    //             .signWith(privateKey)
+    //             .setIssuedAt(new Date(now - (60 * 1000)))
+    //             .setExpiration(new Date(now + (10 * 60 * 1000)))
+    //             .setIssuer(GITHUB_CLIENT_ID)
+    //             .compact();
+    // }
```

Comment:
commented-out code, better to omit?

Reply:
It's a correct calculation, but it turned out to be not needed for our use case